src/main/java/spoon/support/reflect/code/CtVariableAccessImpl.java

```diff
@@ -58,6 +58,9 @@
 	@Override
 	public <C extends CtTypedElement> C setType(CtTypeReference<T> type) {
 		if (type != null) {
+			type.setParent(this);
```

Comment:
Then the contract should not be respected for this specific setter. I could simply ignore it, but I imagine in the future it won't be the only one. 
So should I create a specific annotation to mark setters which does not respect this contract ? Or do you have another idea ?